src/erc-20/erc-20-permit.ts

```diff
@@ -194,26 +194,26 @@ const permitInterface = new Interface([
  * ### Encodes permit data for given erc20 contract and options
  * @param erc20 Erc20 contract instance of contract address
  */
-export const encodePermit =
-  (erc20: Erc20 | Address) =>
-  (options: PermitOptions): string => {
-    const [functionName, amount, deadline] = isAllowedPermit(options)
-      ? ['selfPermitAllowed', options.nonce, options.expiry]
-      : ['selfPermit', options.amount, options.deadline]
+export const encodePermit = (erc20: Erc20 | Address) => (
```

Comment:
Do we really need Address here? I can only see it being used with Contract types being passed in like [here](https://github.com/Phuture-Finance/phuture-sdk/blob/0d2a2a409659d4161939e5abd14f5081a26bfae3/src/erc-20/erc-20-permit.ts#L219)

Reply:
If it is used outside of contract and the address is passed in, then please ignore.